windows-containers/productization.md

```diff
@@ -0,0 +1,180 @@
+* Perform all the required steps within the node for it to be a Kubernetes
+worker
+* Upgrade of all OpenShift related components in the node
+
+### Non-Goals
+
+As part of this enhancement we do not plan to support:
+* Windows node provisioning or de-provisioning
+* Installing the container runtime in the Windows node
+* Windows operating system upgrades
+* Node management like reboots and node draining
+* OpenShift Builds and DeploymentConfigs
+
+## Proposal
+
+The two main pieces that will allow us to achieve our goals is a Windows Scale
+Up (WSU) Ansible playbook and a Windows Machine Config Bootstrapper (WMCB)
+binary. The Ansible playbook will collect information from the cluster and
+transfer it to the Windows node. The WMCB will then perform the necessary
+steps on the node for it to join the cluster.
+
+The reason for having a Ansible playbook and on-node executable split is to
+be future proof when we might want to move to an operator workflow. Please read
+the [alternatives](#Alternatives) section.
+
+### Implementation Details
+
+#### Windows Scale Up (WSU)
+
+The Windows Scale Up is an Ansible playbook that has the follow prerequisites:
+* Needs to run on a Linux system
+* Needs to be able to access the cluster where the Windows worker will be added
+* Needs to be able to access the Windows node
+
+The inputs that the WSU playbook requires are:
+* Windows node credentials
+* Kubelet download location
+* Worker ignition
+* Network configuration (?)
+  * The details of this will be worked out as part of the enhancement
+    proposal for "OVN Plumbing for Hybrid Linux+Windows Clusters GA"
+* WMCB download location
+
+The actions that the WSU will perform are:
+* Check if the container runtime is present on the Windows node
+* Download and copy the kubelet to the Windows node
+* Extract the worker ignition from the MCS pods and copy it to the Windows node
+  * This is an option only for user provisioned clusters. We are investigating
+    alternatives for installer provisioned clusters.
+* Extract and copy the network configuration to the Windows node (?)
+* Download the WMCB, copy it to the Windows node and execute it
+
+#### Windows Machine Config Bootstrapper (WMCB)
+
+The Windows Machine Config Bootstrapper executable has the following
+prerequisites:
+* Only supports Windows x86-64 nodes (Windows Server 2019)
+* Can only run on Windows nodes
+
+The inputs that the WMCB requires are:
+* kubelet location on the local disk
+* Worker ignition location on the local disk
+* Network configuration location on the local disk (?)
+  * The details of this will be worked out as part of the enhancement
+    proposal for "OVN Plumbing for Hybrid Linux+Windows Clusters GA"
+
+The actions that the WMCB will perform are:
+* Install and configure kubelet
+* Parse the worker ignition and extract the bootstrap kubeconfig and the
+kubelet configuration
+  * This is the only use case for the ignition file
```

Comment:
Ignition will not be used to transport anything out of windows node.(atleast this is what I understood my Scott's comments yesterday, better to get some clarity from him)

Reply:
I thought it was to transport into the Windows node. Doesn't my point cover both cases by stating this is only way we will use ignition?

Reply:
Ya, I meant not out of the node where MCS is running(into the windows is right term) - I think we need to call out that we're not using ignition during node booting staging but we're just using the ignition file to parse and get information needed to set up kubelet

Reply:
So you want me to replace "This is the only use case for the ignition file" with "We are not using ignition file during the node booting stage"?

Reply:
Either of them look good. I'd prefer the 2nd one.

Reply:
Either of them work, I'd prefer the 2nd one